python/cudf/cudf/core/series.py

```diff
@@ -976,14 +975,19 @@ def __repr__(self):
         else:
             output = preprocess.to_pandas().__repr__()
         lines = output.split("\n")
-        if isinstance(preprocess._column, cudf.core.column.CategoricalColumn):
-            for idx, value in enumerate(preprocess):
-                if value is None:
-                    lines[idx] = lines[idx].replace(" NaN", "null")
-        if is_datetime_dtype(preprocess.dtype):
-            for idx, value in enumerate(preprocess):
-                if value is None:
-                    lines[idx] = lines[idx].replace(" NaT", "null")
+
+        if preprocess.null_count:
+            if isinstance(
+                preprocess._column, cudf.core.column.CategoricalColumn
+            ):
+                for idx in range(len(preprocess)):
+                    if preprocess[idx] is None:
+                        lines[idx] = lines[idx].replace(" NaN", "null")
+
+            if is_datetime_dtype(preprocess.dtype):
+                for idx in range(len(preprocess)):
+                    if preprocess[idx] is None:
+                        lines[idx] = lines[idx].replace(" NaT", "null")
```

Comment:
Instead of doing all of this craziness, we should be able to use `pd.Series.to_string(na_rep="null")` and it should simplify things quite a bit. The key would be differentiating between `NaN` and `null` in float columns, but we should be able to convert those to object types to handle that.